File: 3542-minimum-operations-to-convert-all-elements-to-zero/3542-minimum-operations-to-convert-all-elements-to-zero.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        #counter of operations
        operations = 0
        stack = [0]
        #Iterate through array
        for num in nums:
            while num < stack[-1]:
                stack.pop()

            if num > stack[-1]:
                operations += 1
                stack.append(num)
        #return number of operations
        return operations

        

File: 3542-minimum-operations-to-convert-all-elements-to-zero/3542-minimum-operations-to-convert-all-elements-to-zero.java
class Solution {
    public int minOperations(int[] nums) {
        Deque<Integer> stack = new ArrayDeque<>();

        int operations = 0;
        stack.push(0);

        for (int num: nums){
            while(!stack.isEmpty() && stack.peek() > num) {
                stack.pop();
            }
            if (stack.peek() < num){
                operations++;
                stack.push(num);
            }
        }
        return operations;
    }
}